core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1018,7 +1025,9 @@ private[spark] class BlockManager(
           try {
             replicate(blockId, bytesToReplicate, level, remoteClassTag)
           } finally {
-            bytesToReplicate.dispose()
+            if (!level.useOffHeap) {
```

Comment:
As `StorageUtils.dispose` only cleans up a memory-mapped `ByteBuffer`, I don't think calling `bytesToReplicate.dispose()` here would be a problem.